application/forms/EventRuleConfigElements/EscalationCondition.php

```diff
@@ -0,0 +1,283 @@
+        }
+
+        if ($removePosition) {
+            unset($this->conditions[$removePosition]);
+            $conditionCount -= 1;
+            if ($conditionCount === 1 && ! $this->allowZeroConditions && $removePosition === 2) {
+                $this->conditions[1]->removeButton = null;
+            } else {
+                for ($n = $removePosition; $n <= $conditionCount; $n++) {
+                    $nextCount = $n + 1;
+                    $this->conditions[$nextCount]->conditionType->setName('column_' . $n);
+                    $this->conditions[$nextCount]->operator->setName('operator_' . $n);
+                    $this->conditions[$nextCount]->conditionVal->setName('val_' . $n);
+                    if ($conditionCount === 1 && ! $this->allowZeroConditions) {
+                        $this->conditions[$nextCount]->removeButton = null;
+                    } elseif ($this->conditions[$nextCount]->removeButton) {
+                        $this->conditions[$nextCount]->removeButton->setSubmitValue((string) $n);
+                    }
+                }
+            }
+
+            $this->getElement('condition-count')->setValue($conditionCount);
+        }
+
+        if ((int) $conditionCount === 0) {
+            $this->addAttributes(['class' => 'zero-escalation-condition']);
+        } elseif ($this->getAttributes()) {
+            $this->getAttributes()->remove('class', 'zero-escalation-condition');
+        }
+
+        $this->add(new EscalationConditionList($this->conditions));
+
+        $this->addElement($addCondition);
+    }
+
+    /**
+     * Create remove button for the condition in the given position
+     *
+     * @param int $count
+     *
+     * @return ?SubmitButtonElement
+     */
+    protected function createRemoveButton(int $count): ?SubmitButtonElement
+    {
+        // check for count and if allow zero conditions
+        $conditionCount = $this->getValue('condition-count');
+        if ((int) $conditionCount === 1 && ! $this->allowZeroConditions) {
+            return null;
+        }
+
+        $removeButton = new SubmitButtonElement(
+            'remove',
+            [
+                'class'          => ['remove-button', 'control-button', 'spinner'],
+                'label'          => new Icon('minus'),
+                'title'          => $this->translate('Remove'),
+                'formnovalidate' => true,
+                'value'          => (string) $count
+            ]
+        );
+
+        $this->registerElement($removeButton);
+
+        return $removeButton;
+    }
+
+    public function hasValue(): bool
+    {
+        $this->ensureAssembled();
+        return parent::hasValue();
+    }
+
+    /**
+     * Get the rendered condition
+     *
+     * @return string
+     */
+    public function getCondition(): string
+    {
+        $filter = Filter::any();
+        /** @var int $count */
+        $count = $this->getValue('condition-count');
```

Comment:
Casting to int is required while after autosubmit the values that are populated from the post request and this would be a string after the autosubmit, but as you said casting back to string is redundant.